esi_leap/resource_objects/ironic_node.py

```diff
@@ -59,30 +59,36 @@ def get_node_config(self):
 
     def set_contract(self, contract):
         patches = []
-        if contract is None:
-            if self.get_contract_uuid():
-                patches.append({
-                    "op": "remove",
-                    "path": "/properties/contract_uuid",
-                })
-            if self.get_project_id():
-                patches.append({
-                    "op": "remove",
-                    "path": "/lessee",
-                })
-        else:
+        patches.append({
+            "op": "add",
+            "path": "/properties/contract_uuid",
+            "value": contract.uuid,
+        })
+        patches.append({
+            "op": "add",
+            "path": "/lessee",
+            "value": contract.project_id,
+        })
+        if len(patches) > 0:
+            get_ironic_client().node.update(self._uuid, patches)
+
+    def expire_contract(self, contract):
+        patches = []
```

Comment:
Now that we can expire a specific contract, it may make sense to add something like:
```
if self.get_contract_uuid() != contract.uuid:
   return
```
This way there's no chance of accidentally expiring a node whose contract as already been assigned to a new contract.

Reply:
Got it!